### PR TITLE
chore(go.mod): bump go-header

### PR DESCRIPTION
**Pull request by renaynay**
Bumps go-header to the latest RC to pull in the fix to the `peerTracker` in https://github.com/celestiaorg/go-header/pull/139. It's suspected that the `peerTracker`'s aggressive scoring mechanism is causing header syncing to stall after periods of poor connectivity.

### CODE REVIEW

**Review by ramin — APPROVED** [review]
:shipit:

**Review by walldiss — APPROVED** [review]

**Comment by renaynay** [comment]
I forgot metrics broke in this bump

**Review by ramin — APPROVED** [review]

File: nodebuilder/header/constructors.go [hunk 1]
```diff
@@ -99,16 +99,14 @@ func newInitStore[H libhead.Header[H]](
 	ds datastore.Batching,
 	ex libhead.Exchange[H],
 ) (libhead.Store[H], error) {
-	s, err := store.NewStore[H](ds, store.WithParams(cfg.Store))
-	if err != nil {
-		return nil, err
+	opts := []store.Option{store.WithParams(cfg.Store)}
+	if MetricsEnabled {
+		opts = append(opts, store.WithMetrics())
```
Comment (walldiss) [inline nodebuilder/header/constructors.go:104]: Looks like we have two different patters for initialising metrics now. Do you know why?

**Review by walldiss — APPROVED** [review]

**Review by distractedm1nd — APPROVED** [review]
Same question as Vlad

**Pull request merged**